src/main/java/com/endava/endabank/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,236 @@
+import java.util.Set;
+import java.util.function.BiFunction;
+
+@Service
+@AllArgsConstructor
+public class UserServiceImpl implements UserService {
+    private UserDao userDao;
+    private ModelMapper modelMapper;
+    private IdentifierTypeService identifierTypeService;
+    private RoleService roleService;
+    private PasswordEncoder passwordEncoder;
+    private ForgotUserPasswordTokenService forgotUserPasswordTokenService;
+    private MailService mailService;
+    private MailProperties mailProperties;
+    @Override
+    @Transactional(readOnly = true)
+    public List<UserToApproveAccountDto> usersToApprove() {
+        List<User> users = userDao.findAll();
+        return users.stream().map(this::mapToUserToApproveDto).toList();
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public User findById(Integer id) {
+        return userDao.findById(id).
+                orElseThrow(() -> new UsernameNotFoundException(Strings.USER_NOT_FOUND));
+    }
+
+    @Override
+    @Transactional
+    public User save(UserRegisterDto userDto) {
+        User user = modelMapper.map(userDto, User.class);
+        Optional<User> userExist = userDao.findByEmail(user.getEmail());
+        if (userExist.isPresent()) {
+            throw new UniqueConstraintViolationException(Strings.CONSTRAINT_EMAIL_VIOLATED);
+        }
+        userExist = userDao.findByIdentifier(user.getIdentifier());
+        if (userExist.isPresent()) {
+            throw new UniqueConstraintViolationException(Strings.CONSTRAINT_IDENTIFIER_VIOLATED);
+        }
+        Role role = roleService.findById(Permissions.ROLE_USER);
+        IdentifierType identifierType = identifierTypeService.findById(userDto.getTypeIdentifierId());
+        user.setRole(role);
+        user.setIdentifierType(identifierType);
+        user.setPassword(passwordEncoder.encode(user.getPassword()));
+        userDao.save(user);
+        return user;
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public UsernamePasswordAuthenticationToken getUsernamePasswordToken(Integer userId) {
+        User user = this.findById(userId);
+        Role role = user.getRole();
+        Set<Permission> permissions = role.getPermissions();
+        List<SimpleGrantedAuthority> authorities = new ArrayList<>();
+        authorities.add(new SimpleGrantedAuthority(role.getName()));
+        permissions.forEach(permission -> authorities.add(new SimpleGrantedAuthority(permission.getName())));
+        UserPrincipalSecurity userPrincipalSecurity = modelMapper.map(user, UserPrincipalSecurity.class);
+        return
+                new UsernamePasswordAuthenticationToken(userPrincipalSecurity, null, authorities);
+    }
+    @Override
+    @Transactional
+    public UserToApproveAccountDto updateUserAccountApprove(Integer id, boolean value) {
+        User user = this.findById(id);
+        user.setIsApproved(value);
+        userDao.save(user);
+        return this.mapToUserToApproveDto(user);
```

Comment:
Hi,

If you return this, you are gonna return the Dto based on the user of line 115 instead of the saved user.